addons/resource/models/resource.py

```diff
@@ -2,46 +2,35 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import datetime
+from operator import itemgetter
+
 from dateutil import rrule
 from dateutil.relativedelta import relativedelta
-from openerp.tools import DEFAULT_SERVER_DATE_FORMAT
-from operator import itemgetter
 
-from openerp import tools
-from openerp.osv import fields, osv
-from openerp.tools.float_utils import float_compare
-from openerp.tools.translate import _
-from openerp.exceptions import UserError
-import pytz
+from odoo import api, fields, models, _
+from odoo.tools.float_utils import float_compare
+from odoo.exceptions import UserError, ValidationError
 
-class resource_calendar(osv.osv):
-    """ Calendar model for a resource. It has
 
+class ResourceCalendar(models.Model):
+
+    """ Calendar model for a resource. It has
      - attendance_ids: list of resource.calendar.attendance that are a working
                        interval in a given weekday.
      - leave_ids: list of leaves linked to this calendar. A leave can be general
                   or linked to a specific resource, depending on its resource_id.
-
     All methods in this class use intervals. An interval is a tuple holding
     (begin_datetime, end_datetime). A list of intervals is therefore a list of
     tuples, holding several intervals of work or leaves. """
     _name = "resource.calendar"
     _description = "Resource Calendar"
 
-    _columns = {
-        'name': fields.char("Name", required=True),
-        'company_id': fields.many2one('res.company', 'Company', required=False),
-        'attendance_ids': fields.one2many('resource.calendar.attendance', 'calendar_id', 'Working Time', copy=True),
-        'manager': fields.many2one('res.users', 'Workgroup Manager'),
-        'leave_ids': fields.one2many(
-            'resource.calendar.leaves', 'calendar_id', 'Leaves',
-            help=''
-        ),
-    }
-    _defaults = {
-        'company_id': lambda self, cr, uid, context: self.pool.get('res.company')._company_default_get(cr, uid, 'resource.calendar', context=context)
-    }
-
+    name = fields.Char(required=True)
+    company_id = fields.Many2one('res.company', string='Company',
+        default=lambda self: self.env['res.company']._company_default_get('resource.calendar'))
```

Comment:
_company_default_get method returns the default company (the user's company) but now 'object' and 'field' arguments are ignored. instead of this method directly use current user's company like : 
default= lambda self: self.env.user.company_id